src/sitemap-pipe.js

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import { cleanupHeaderValue } from '@adobe/helix-shared-utils';
+import { authenticate, requireProject } from './steps/authenticate.js';
+import fetchConfig from './steps/fetch-config.js';
+import fetchContent from './steps/fetch-content.js';
+import fetchConfigAll from './steps/fetch-config-all.js';
+import renderCode from './steps/render-code.js';
+import setXSurrogateKeyHeader from './steps/set-x-surrogate-key-header.js';
+import setCustomResponseHeaders from './steps/set-custom-response-headers.js';
+import { PipelineStatusError } from './PipelineStatusError.js';
+import { PipelineResponse } from './PipelineResponse.js';
+
+/**
+ * Serves or renders the sitemap xml. The sitemap is always served from the preview content-bus
+ * partition.
+ *
+ * todo: currently only serves an existing sitemap.xml from the contentbus.
+ *       generate sitemap on the fly based on the sitemap.json
+ *
+ * @param {PipelineState} state
+ * @param {PipelineRequest} req
+ * @returns {PipelineResponse}
+ */
+export async function sitemapPipe(state, req) {
+  const { log } = state;
+  state.type = 'sitemap';
+  // force loading from preview
+  state.partition = 'preview';
+
+  if (state.info?.path !== '/sitemap.xml') {
+    // this should not happen as it would mean that the caller used the wrong route. so we respond
+    // with a 500 to indicate that something is wrong.
+    return new PipelineResponse('', {
+      status: 500,
```

Comment:
400 or 404 instead of 500?

Reply:
I think 500 is OK, because it indicates that we let some request go through Fastly/CF that shouldn't